### PR TITLE
Sorting & Refactoring

### PR DESCRIPTION
**Pull request by AbdullatifGhajar**

### CODE REVIEW

File: src/code_translation/NodeSelector.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def resolve_Tuple(self, node: cst.Tuple) -> Sequence:
         return tuple(ret_values)
 
     def resolve_SimpleString(self, node: cst.SimpleString) -> str:
-        return node.value
+        return node.evaluated_value
```
Comment (AbdullatifGhajar) [inline src/code_translation/NodeSelector.py:170]: this will save us a bunch of .strip("'")

File: src/code_translation/ir/nodes.py [hunk 2]
```diff
@@ -8,8 +9,11 @@
 
 
 def str_code_to_cst(code: str) -> cst.CSTNode:
-    cst_tree = cst.parse_expression(code)
-    return cst_tree
+    return cst.parse_expression(code)
+
+
+def evaluate_cst(node: cst.CSTNode) -> any:
```
Comment (AbdullatifGhajar) [inline src/code_translation/ir/nodes.py:15]: if we have some cstnode that we need to evaluate, like a list of strings

File: src/evaluation/examples/join_examples.py [hunk 3]
```diff
@@ -1,19 +1,4 @@
...
-    name: str
-    code: PipelineCode
-    optimized_code: PipelineCode
-
-    def __init__(self, name: str, code: PipelineCode, optimized_code: PipelineCode):
-        self.name = name
-        self.code = dedent(code)
-        self.optimized_code = dedent(optimized_code)
-
+from .pipeline_example import PipelineExample
```
Comment (AbdullatifGhajar) [inline src/evaluation/examples/join_examples.py:1]: don't rely on github. I moved all tests to individual files. I didn't remove tests

File: test/test_orchestrator.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from src.code_translation.orchestrator import Orchestrator
```
Comment (AbdullatifGhajar) [inline test/test_orchestrator.py:1]: This is just a template for further tests for the orchestrator. If we have time and fun we can make the project more maintainable.

**Review by dkuska — APPROVED** [review]

File: src/code_translation/NodeSelector.py (hunk 1 above)
Comment (t-straeubig) [reply src/code_translation/NodeSelector.py:170]: thanks

File: src/code_translation/input/pandas.py [hunk 5]
```diff
@@ -22,24 +29,30 @@ def visit_read_sql(self, *args, **kwargs):
 
     # df methods
 
+    def visit_df_sort_values(self, df_node: DataFrameNode, *args, **kwargs):
+        return SortNode(df_node, *args, **kwargs)
+
     def visit_df_join(self, df_node: DataFrameNode, *args, **kwargs):
+        # TODO This is only if sort is kwargs. Handle if sort is in *args
```
Comment (t-straeubig) [inline src/code_translation/input/pandas.py]: Ich denke hier würde helfen die signatur zu 
def vist_df_join(self, other: DataFrameNode, on=None, how='left', lsuffix='', rsuffix='', sort=False, validate=None)
(also genau der signatur von DataFrame.join)
zu verändern.

File: src/code_translation/input/pandas.py [hunk 6]
```diff
@@ -22,24 +29,30 @@ def visit_read_sql(self, *args, **kwargs):
...
+        if kwargs.get("sort", False):
+            return SortNode(JoinNode(df_node, *args, **kwargs))
         return JoinNode(df_node, *args, **kwargs)
 
     def visit_df_merge(self, df_node: DataFrameNode, *args, **kwargs):
+        if kwargs.get("sort", False):
```
Comment (t-straeubig) [inline src/code_translation/input/pandas.py]: Hier auch lieber signatur ändern

File: src/code_translation/ir/nodes.py [hunk 7]
```diff
@@ -70,6 +76,66 @@ def to_cst_translation(self, sql_access_method) -> CSTTranslation:
...
+    ):
+        node.parent = self
+        self.node = node
+        if isinstance(by, cst.CSTNode):
+            by = evaluate_cst(by)
+        if isinstance(by, str):
+            by = [by]
+        self.by = by
+        if isinstance(ascending, cst.CSTNode):
+            ascending = evaluate_cst(ascending)
```
Comment (t-straeubig) [inline src/code_translation/ir/nodes.py:96]: Das verstehe ich nicht ganz. by ist laut type hint der Methodensignatur immer ein str, ascending ist immer boolean. In diesem Fall machen die Abfragen keinen Sinn

**Review by t-straeubig — APPROVED** [review]

File: src/code_translation/ir/nodes.py (hunk 7 above)
Comment (AbdullatifGhajar) [reply src/code_translation/ir/nodes.py:96]: by in pandas.sort_values could be a string or a list of str. So, I need to convert it to a list of strings. Sometimes it is unresolved, which means I get a CSTNode, so I need to evaluate it. I know it isn't nice, but it is not new. We can refactor this and all the other arguments later.

**Pull request merged**